compiler-core/src/requirement.rs

```diff
@@ -2,22 +2,48 @@ use std::fmt;
 use std::str::FromStr;
 
 use crate::error::Result;
-use crate::io::make_relative;
+use crate::io::{is_same_file, make_relative};
 use camino::{Utf8Path, Utf8PathBuf};
 use ecow::EcoString;
 use hexpm::version::Range;
 use serde::de::{self, Deserializer, MapAccess, Visitor};
 use serde::ser::{Serialize, SerializeMap, Serializer};
 use serde::Deserialize;
 
-#[derive(Deserialize, Debug, PartialEq, Eq, Clone)]
+#[derive(Deserialize, Debug, Eq, Clone)]
 #[serde(untagged, remote = "Self")]
 pub enum Requirement {
     Hex { version: Range },
     Path { path: Utf8PathBuf },
     Git { git: EcoString },
 }
 
+impl PartialEq for Requirement {
+    fn eq(&self, other: &Self) -> bool {
+        match (self, other) {
+            (Self::Path { path: own_path }, Self::Path { path: other_path }) => {
+                is_same_file(own_path, other_path).unwrap_or(false)
```

Comment:
We don't want to do take symlinks etc into consideration so we should not use this library. Instead we can use the same logic that is used to make the absolute path in the manifest.

https://github.com/gleam-lang/gleam/blob/4f6ba0a88b5c3589636637721460dfd47c6b44e1/compiler-cli/src/dependencies.rs#L701-L705

Reply:
Thanks! `is_same_requirements` sounds fab to me, and easy to test that function. Thank you.